python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2394,6 +2399,71 @@ def get_handlers(self) -> Sequence[logging.Handler]:
     def store_event(self, event: "EventLogEntry") -> None:
         self._event_storage.store_event(event)
 
+    def should_retry_run(self, run: DagsterRun) -> bool:
+        """Determines if a run will be retried by the automatic reexcution system.
+        A run will retry if:
+        - it is failed.
+        - the number of max allowed retries is > 0 (max retries can be set via system setting or run tag).
+        - there have not already been >= max_retries retries for the run.
+
+        If the run failure reason was a step failure and the retry_on_asset_or_op_failure tag/system setting is set to false,
+        a warning message will be logged and the run will not be retried.
+
+        We determine how many retries have been launched for the run by looking at the size of the run group
+        (the set of runs that have the same root_run_id and the run with root_run_id). Since manually launched retries are
+        given a root_run_id, this means that if a user launches a manual retry of run A and then this function
```

Comment:
this isn't saying that *only* manually launched retries are given a root_run_id right (instead, all retried runs are) - that's how I parsed it at first

Reply:
it's not saying that - i'll clarify